apps/gpc-verifier/app/util/persistence.ts

```diff
@@ -0,0 +1,17 @@
+import { kv } from "@vercel/kv";
+
+// Saves the set of nullifier hashes we have seen
+const NULLIFIER_KV_SET = "nullifiers_hash";
+
+// Checks whether we've seen the nullifier hash before.
```

Comment:
Is there a way to reset this for testing?
Maybe the instructions for that should be part of the error message or tooltip when a user fails to apply twice.

Reply:
Added a new issue to track this.